src/frontend/query_commands.ml

```diff
@@ -512,6 +512,29 @@ let dispatch pipeline (type a) : a Query_protocol.t -> a =
     | Some res -> `Found res 
     | None -> `No_documentation)
 
+  | Expand_node pos -> 
+    let pos = Mpipeline.get_lexing_pos pipeline pos in
+    let typer = Mpipeline.typer_result pipeline in 
+    let node = Mtyper.node_at typer pos in
+    let check_ppx (node: Env.t * Mbrowse.node) : bool =
+      let _,n = node in
+      let has_a_deriver =
+        Browse_raw.has_attr ~name:"deriving" n in
+      let attrs = Browse_raw.node_attributes n in
+      List.exists ~f:(fun (a:Typedtree.attribute) ->
+        let loc = a.attr_loc in
+            loc.loc_start.pos_cnum <= pos.pos_cnum
+            && loc.loc_end.pos_cnum >= pos.pos_cnum
+      ) attrs && has_a_deriver
+    in 
+    let has_deriver = List.exists ~f:check_ppx node in
+    if has_deriver then 
+      let root = (List.hd (List.rev node)) in
```

Comment:
You're going all the way to the top-level structure/signature here. Instead, you'll want to take the structure/signature that the structure/signature item belongs to that the deriver is attached to. Let me explain with an example:
```
module M = struct
    module N = struct
       type t = int [@@deriving foo]
       let a = 1
    end
    let b = 2
end
let c = 3
```

In that example, `root` will contain the module binding of `M` and the value binding of `c`. And you'll retrieve all items from `root` that contain the cursor position, i.e. you'll retreive the whole module binding of `M`. Instead, you'll want `root` to be the structure of the module expression of `N`, i.e. the type declaration of `t`, the derived structure items, and the value binding of `a`. That way, when you retrieve all items from `root` that contain the cursor position, you'll get the type definition of `t` and the derived code from it.

Let me know if it's clear what I mean!

Reply:
Now, it will only return the exact ppx node which has been added. I am adding a conditional `loc_ghost = true` to make sure we get only the `include ...` part of it.